### PR TITLE
feat(#158): Convert units to CO2 or $

### LINKED ISSUE #158: Convert energy to CO2 or $
More in depth than #54, I’d like to display my energy consumption, including both electricity and fossil fuel. Specifically, I’d like a sankey in which a column has three entries: low-carbon electricity, fossil electricity, and natural gas. Or perhaps the first column has “clean energy” and “fossil energy” which redistribute into “electricity” and “gas” in the second column. Either way, my dozens of electricity-monitoring entities would feed from there.

While kJ could be used as a common unit, I am more interested in two other common units: dollars, using the same price of energy that the Energy Dashboard does, and CO2eq, using the same CO2/kWh carbon intensity that the Energy Dashboard does.

The result would be that I can set the energy time duration to the last month and see the cost and carbon intensity of every entity - including both my heat pump and gas furnace.

### PR DESCRIPTION
**Pull request by madsciencetist**
A sankey chart used to display energy usage (gas and/or electric) can now convert the native energy units (kWh or cubic feet of gas) into their corresponding monetary cost or CO2 intensity.

**Cost:**
```
energy_date_selection: true
convert_units_to: monetary
monetary_unit: USD
gas_price: 0.022405
electricity_price: 0.30112
```
image
Note that while HA's Energy Dashboard knows and tracks utility prices, the API does not make them available to us outside of over-summarizing statistics like total cost. Thus utility cost must be supplied to the sankey config, and only fixed rates are supported.

**CO2 intensity:**
```
energy_date_selection: true
convert_units_to: gCO2
unit_prefix: k
```
image

For carbon intensity, we are able to use (abuse) the HA API to actually get the carbon intensity of all electricity usage on an hourly basis. Gas is assumed to have a fixed (but configurable) carbon intensity of 66.6 gCO2 per cubic foot.

Closes #158

### CODE REVIEW

**Comment by madsciencetist** [comment]
This is my first time using lovelace, TS or JS, so feel free to correct what doesn't make sense.

File: src/section.ts [hunk 1]
```diff
@@ -152,8 +152,8 @@ export function renderSection(props: {
             ${shouldShowLabel
               ? html`<div class="label" style=${styleMap(labelStyle)}>
                   ${show_states && isNotPassthrough
-                    ? html`<span class="state">${formattedState}</span>${show_units
-                          ? html`<span class="unit">${box.unit_of_measurement}</span>`
+                    ? html`<span class="state">${formattedState}</span>${(show_units && convert_units_to != 'monetary')
+                          ? html`&nbsp;<span class="unit">${box.unit_of_measurement}</span>`
```
Comment (MindFreeze) [inline src/section.ts:156]: This space and the one above shouldn't be part of this PR. The convention is that unit of measurement is added without space and users can add space with card-mod and CSS if they wish.

**Review by MindFreeze — CHANGES_REQUESTED** [review]

File: src/section.ts (hunk 1 above)
Comment (madsciencetist) [reply src/section.ts]: Fixed

File: src/ha-sankey-chart.ts [hunk 2]
```diff
@@ -111,7 +112,14 @@ class SankeyChart extends SubscribeMixin(LitElement) {
             }
           }
           if (this.entityIds.length) {
-            const stats = await getStatistics(this.hass, data, this.entityIds);
+            const conversions: Conversions = {
```
Comment (MindFreeze) [inline src/ha-sankey-chart.ts:115]: Please move the default values from here to `normalizeConfig` in `utils.ts`. I'd like to have all the  default values in one place

Comment (madsciencetist) [reply src/ha-sankey-chart.ts:115]: Done

**Review by MindFreeze — APPROVED** [review]
